tck/src/main/java/org/eclipse/microprofile/lra/tck/TckTests.java

```diff
@@ -462,6 +467,137 @@ public void noLRATest() throws WebApplicationException {
+     * cancel the LRA
+     * check that both participants were asked to compensate
+     */
+    @Test
+    public void joinWithTwoResourcesWithCancel() throws WebApplicationException {
+        joinWithTwoResources(false);
+    }
+
+    private void joinWithOneResource(String methodName, boolean close, String resouce1Method, String resouce2Method)
+            throws WebApplicationException {
+        // set up the web target for the test
+        WebTarget resource1Path = tckSuiteTarget.path(TCK_PARTICIPANT_RESOURCE_PATH).path(resouce1Method);
+        WebTarget resource2Path = tckSuiteTarget.path(TCK_PARTICIPANT_RESOURCE_PATH).path(resouce2Method);
+
+        URI lra = lraClient.startLRA(null, lraClientId(), lraTimeout(), ChronoUnit.MILLIS);
+
+        // get the initial values for the number of completions and compensations
+        int completedCount = getActivityCount(TCK_PARTICIPANT_RESOURCE_PATH, COMPLETED_CNT_PATH);
+        int compensatedCount = getActivityCount(TCK_PARTICIPANT_RESOURCE_PATH, COMPENSATED_CNT_PATH);
+
+        // invoke the same JAX-RS resources twicein the context of the lra which should enlist the resource only once:
+        Response response1 = resource1Path.request().header(LRA.LRA_HTTP_HEADER, lra).put(Entity.text(""));
+        checkStatusAndClose(Response.Status.OK, response1, resource1Path);
+        Response response2 = resource2Path.request().header(LRA.LRA_HTTP_HEADER, lra).put(Entity.text(""));
+        checkStatusAndClose(Response.Status.OK, response2, resource2Path);
+
+        if (close) {
+            lraClient.closeLRA(lra);
+
+            assertTrue(methodName + ": resource should have completed once with no compensations",
+                    completedCount + 1 == getActivityCount(TCK_PARTICIPANT_RESOURCE_PATH, COMPLETED_CNT_PATH)
+                            && compensatedCount == getActivityCount(TCK_PARTICIPANT_RESOURCE_PATH, COMPENSATED_CNT_PATH));
+        } else {
+            lraClient.cancelLRA(lra);
+
+            assertTrue(methodName + ":: resource should have compensated once with no completions",
+                    completedCount == getActivityCount(TCK_PARTICIPANT_RESOURCE_PATH, COMPLETED_CNT_PATH)
+                            && compensatedCount + 1 == getActivityCount(TCK_PARTICIPANT_RESOURCE_PATH, COMPENSATED_CNT_PATH));
+        }
+    }
+
+    private void joinWithTwoResources(boolean close) throws WebApplicationException {
+        // set up the web target for the test
+        WebTarget resource1Path = tckSuiteTarget.path(LRA_CONTROLLER_PATH).path(TRANSACTIONAL_WORK_PATH);
+        WebTarget resource2Path = tckSuiteTarget.path(TCK_PARTICIPANT_RESOURCE_PATH).path(JOIN_WITH_EXISTNG_LRA_PATH);
+
+        URI lra = lraClient.startLRA(null, lraClientId(), lraTimeout(), ChronoUnit.MILLIS);
+
+        // get the initial values for the number of completions and compensations
+        int completedCount1 = getActivityCount(LRA_CONTROLLER_PATH, "completedactivitycount");
```

Comment:
I would put the retrieval of the initial values first before any interaction with Participant/coordincator so that we are absolutely sure about the interactions.